audio-worklet/src/noisegate-script-processor.js

```diff
@@ -108,11 +108,11 @@ class NoiseGate {
 
     let envelope = this.detectLevel_(this.channel_);
     envelope = this.convertEnvelopeToSineNormalizedDecibelScale_(envelope);
-
+    let weights = this.computeWeights_(envelope);
```

Comment:
Ok. Always a tricky question:  How much should the internal implementation be modified to support testing and how much should being able to test influence the implementation.